addons/test_mail/tests/test_mail_mail.py

```diff
@@ -823,6 +830,97 @@ def test_mail_mail_values_unicode(self):
         self.assertEqual(self._mails[0]['email_cc'], ['test.😊.cc@example.com'])
         self.assertEqual(self._mails[0]['email_to'], ['test.😊@example.com'])
 
+    @mute_logger('odoo.addons.mail.models.mail_mail')
+    def test_mail_mail_send_server_attachment_to_download_link(self):
+        """ Test that when the mail size exceeds the max email size limit,
+        attachments are turned into download links added at the end of the
+        email content.
+
+        The feature is tested in the following conditions:
+        - using a specified server or the default one (to test command ICP parameter)
+        - in batch mode
+        - with mail that exceed (with one or more attachments) or not the limit
+        - with attachment owned by a business record or not: attachments not owned by a
+        business record are never turned into links because their lifespans are not
+        controlled by the user (might even be deleted right after the message is sent).
+        """
+        def count_attachments(message):
+            return (0 if isinstance(message, str)
+                    else sum(count_attachments(part) for part in message.get_payload()) if message.is_multipart()
+                    else 'attachment' in message.get('Content-Disposition', ''))
+
+        attachment_size_in_bytes = 1024 * 128
+        datas = base64.b64encode((''.join('.' for _ in range(0, attachment_size_in_bytes))).encode())
```

Comment:
Huh? Are you encoding about 128.000 of "." characters in your test? Seems like a bad idea ^^

Can't we instead either mock "_estimate_mail_size", or better yet mock the "file_size" field?